src/moonlink/src/storage/mooncake_table/snapshot_read_output.rs

```diff
@@ -71,50 +84,115 @@ impl ReadOutput {
             .unwrap();
     }
 
+    /// File ordering should be preserved because the index determines the position
+    /// in the final [`resolved_data_files`] array, which delete or other operations rely on.
```

Comment:
Documentation incomplete: since we do special handling on errors encountered, you need to clearly document we unpin cache handles, so callers don't need to unpin one more time.